components/camera/client_test.go

```diff
@@ -366,12 +388,109 @@ func TestClient(t *testing.T) {
 		ext := data.FromDMExtraMap
 		ext["hello"] = "world"
 		ctx = context.Background()
-		_, _, err = camClient.Images(ctx, ext)
+		_, _, err = camClient.Images(ctx, nil, ext)
 		test.That(t, err, test.ShouldNotBeNil)
 		test.That(t, err.Error(), test.ShouldContainSubstring, errGetImageFailed.Error())
 
 		test.That(t, conn.Close(), test.ShouldBeNil)
 	})
+
+	t.Run("camera client images filter source names", func(t *testing.T) {
+		ctx := context.Background()
+		conn, err := viamgrpc.Dial(context.Background(), listener1.Addr().String(), logger)
+		test.That(t, err, test.ShouldBeNil)
+		defer func() {
+			test.That(t, conn.Close(), test.ShouldBeNil)
+		}()
+
+		camClient, err := camera.NewClientFromConn(context.Background(), conn, "", camera.Named(testCameraName), logger)
+		test.That(t, err, test.ShouldBeNil)
+
+		injectCamera.ImagesFunc = func(
+			ctx context.Context,
+			filterSourceNames []string,
+			extra map[string]interface{},
+		) ([]camera.NamedImage, resource.ResponseMetadata, error) {
+			namedImgColor, err := camera.NamedImageFromImage(expectedColor, "color", rutils.MimeTypeRawRGBA)
+			if err != nil {
+				return nil, resource.ResponseMetadata{}, err
+			}
+			namedImgDepth, err := camera.NamedImageFromImage(expectedDepth, "depth", rutils.MimeTypeRawDepth)
+			if err != nil {
+				return nil, resource.ResponseMetadata{}, err
+			}
+
+			if len(filterSourceNames) == 0 {
+				return []camera.NamedImage{namedImgColor, namedImgDepth}, resource.ResponseMetadata{}, nil
+			}
+
+			images := make([]camera.NamedImage, 0, len(filterSourceNames))
+			for _, src := range filterSourceNames {
+				switch src {
+				case "color":
+					images = append(images, namedImgColor)
+				case "depth":
+					images = append(images, namedImgDepth)
+				default:
+					return nil, resource.ResponseMetadata{}, fmt.Errorf("unknown source name: %s", src)
+				}
+			}
+			return images, resource.ResponseMetadata{}, nil
+		}
+
+		t.Run("successful filtering", func(t *testing.T) {
+			images, _, err := camClient.Images(ctx, []string{"color", "depth"}, nil)
+			test.That(t, err, test.ShouldBeNil)
+			test.That(t, len(images), test.ShouldEqual, 2)
+			test.That(t, images[0].SourceName, test.ShouldEqual, "color")
+			test.That(t, images[1].SourceName, test.ShouldEqual, "depth")
+
```

Comment:
Question: if the filter was in the opposite order `[]string{"depth", "color"}`, would it be expected that `images[0]` is the depth image, and `images[1]` is the color image? Is the order of request required in the order of response?

Reply:
I'll add a test to make that clear